src/corehost/cli/deps_resolver.cpp

```diff
@@ -525,9 +525,9 @@ bool deps_resolver_t::resolve_probe_dirs(
         }
         else
         {
-            // For standalone apps, dotnet.exe will be renamed. Do not use the full package name
+            // For standalone apps, apphost.exe will be renamed. Do not use the full package name
             // because of rid-fallback could happen (ex: CentOS falling back to RHEL)
-            if (ends_with(entry.library_name, _X(".Microsoft.NETCore.App"), false) && entry.asset_name == _X("dotnet"))
+            if (ends_with(entry.library_name, _X(".Microsoft.NETCore.App"), false) && (entry.asset_name == _X("dotnet") || entry.asset_name == _X("apphost")))
```

Comment:
@shellap this was changed from https://github.com/dotnet/core-setup/commit/ea86ceed56522a0701dcd6e8966bc3407baab4ed.